src/components/Articles/List/Card/Card.jsx

```diff
@@ -1,19 +1,51 @@
 import { item, imgDiv, title, author, topic, count, votes, dateTime } from "./Card.module.css";
-
+import { useState, useEffect } from "react";
 import { useNavigate } from "react-router-dom";
+import { voteArticle } from "../../../../api";
 
-const Card = ({ article }) => {
-  const date = new Date(article.created_at).toLocaleDateString();
-  const time = new Date(article.created_at).toLocaleTimeString();
-
+const Card = ({ article, setArticles }) => {
+  const date = new Date(article.created_at);
   const goToArticle = useNavigate();
 
+  const [voteValue, setVoteValue] = useState(0);
+
+  const handleVote = (vote) => {
+    voteArticle(article.article_id, vote);
+
+    setArticles((currArticles) => {
+      const updatedArticles = currArticles.map((item) => {
+        const votes = item.votes + vote;
+        if (item.article_id === article.article_id) {
+          return { ...item, votes };
+        }
+        return item;
+      });
+      return updatedArticles;
+    });
+
+    setVoteValue((current) => {
+      return (current += vote);
+    });
+  };
+
+  const handleUpVote = () => {
+    handleVote(1);
+  };
+
+  const handleDownVote = () => {
+    handleVote(-1);
+  };
```

Comment:
It's great that you have your votes optimistically rendering but could the user have some kind of feedback if the patch doesn't work? If i set to offline and try to upVote it gives me the error in the console but there is nothing for the user to see 🤔